src/cargo/util/toml/mod.rs

```diff
@@ -1347,42 +1646,158 @@ impl TomlManifest {
             project.links.as_deref(),
         )?;
 
+        fn resolve_to_option<'a, T>(
+            features: &Features,
+            value: Option<MaybeWorkspace<T>>,
+            label: &str,
+            inheritable_fields: Option<&'a InheritableFields>,
+            resolve: impl FnOnce(&InheritableFields) -> Option<T>,
+        ) -> CargoResult<Option<T>> {
+            match value {
+                None => Ok(None),
+                Some(mw) => {
+                    match mw.resolve(&features, label, || get_ws(inheritable_fields), resolve) {
+                        Ok(t) => Ok(Some(t)),
+                        Err(e) => Err(e),
+                    }
+                }
+            }
```

Comment:
Am I reading this correctly that this is equivalent of
```rust
value
    .map(|mw| mw.resolve(&features, label, || get_ws(inheritable_fields), resolve))
    .transpose()
```

Reply:
You are reading it correctly and I completely forgot about 
```rust
 .transpose()
```
I think that fits much better than a dedicated function so I will change them all to that!